akka-http2-support/src/test/scala/akka/http/impl/engine/http2/ConnectionSpec.scala

```diff
@@ -0,0 +1,54 @@
+package akka.http.impl.engine.http2
+
+import scala.concurrent.duration._
+import akka.http.impl.util.{AkkaSpecWithMaterializer, ExampleHttpContexts}
+import akka.http.scaladsl.{Http, HttpConnectionContext}
+import akka.http.scaladsl.client.RequestBuilding.Get
+import akka.http.scaladsl.model.HttpResponse
+import akka.http.scaladsl.model.StatusCodes.OK
+import akka.http.scaladsl.server.Directives.complete
+import akka.http.scaladsl.settings.ClientConnectionSettings
+import akka.stream.OverflowStrategy
+import akka.stream.scaladsl.{Keep, Sink, Source}
+import akka.util.ByteString
+import org.scalatest.concurrent.ScalaFutures
+
+class ConnectionSpec extends AkkaSpecWithMaterializer(
```

Comment:
Could we put that into `Http2ClientServerSpec` or is there some context that is different enough to warrant a new spec file?

Reply:
`Http2ClientServerSpec` sets up the (secure) connection in the `TestSetup`.

Adding a test there that does not use `TestSetup` seemed a bit inconsistent.

Alternatively we could extend `TestSetup` to allow both secure and insecure tests. Originally I decided against that because I didn't want to make `TestSetup` more complicated just for one test. Thinking about it now, it might still make sense to add that, to make it easier to run a single test there and observe it with wireshark.

That said, I think there's still reason to keep this `ConnectionSpec`: it can be used to test the internal connection API in more detail. In particular, I noticed while writing this test that when using this API with a simple `Source.single` and `Sink.head`, the graph would complete even though the response was not written yet. I didn't want to get further into that for this PR, but that seems like something that is worth investigating.

Reply:
> `Http2ClientServerSpec` sets up the (secure) connection in the `TestSetup`.
> 
> Adding a test there that does not use `TestSetup` seemed a bit inconsistent.

It's more about grouping tests by functionality so when adding features it's clear where to look for existing tests.

> Alternatively we could extend `TestSetup` to allow both secure and insecure tests. Originally I decided against that because I didn't want to make `TestSetup` more complicated just for one test. Thinking about it now, it might still make sense to add that, to make it easier to run a single test there and observe it with wireshark.

Either that, or don't use TestSetup which would be fine for me as well (as long as there not multiple TestSetups per spec).

> That said, I think there's still reason to keep this `ConnectionSpec`: it can be used to test the internal connection API in more detail. In particular, I noticed while writing this test that when using this API with a simple `Source.single` and `Sink.head`, the graph would complete even though the response was not written yet. I didn't want to get further into that for this PR, but that seems like something that is worth investigating.

Good point, created https://github.com/akka/akka-http/issues/3594 to track this feature.

Reply:
> It's more about grouping tests by functionality so when adding features it's clear where to look for existing tests.

From that perspective I think the separate `ConnectionSpec` is better, as those are tests on a different 'level' after all.

Reply:
I see it more like a settings variation of what's otherwise also tested in Http2ClientServerSpec using a reimplementation of basically the same infrastructure (queues instead of probes). :)